### PR TITLE
Added ability to deny unfiltered historical queries if new config opt…

### LINKED ISSUE #175: Add config option to disallow unfiltered historical queries
https://jira.jpl.nasa.gov/browse/ASK-288

**Is your feature request related to a problem? Please describe.**
NISAR user training stipulates for users to filter on a historical session in OMM so that queries don't bog down the MariaDB query database. We’ve our web hosts and db hosts crash during important activities because of the amount of users sending unfiltered queries. We’d like a way to strictly enforce this instead of solely relying on user training.

**Describe the solution you'd like**
We’d like a config option that, when enabled, makes it so that historical queries are not even sent out if a user is not filtering on a session.

**Testing Instructions**
1. Navigate to a plot while not filtered on a session
2. Observe in network tab that an query not filtered by session is sent

### PR DESCRIPTION
**Pull request by chesterhawkins**
closes #175 

New config option sessions.historicalSessionFilter. denyUnfilteredQueries is used to determine whether an unfiltered historical query should be denied or allowed.

### CODE REVIEW

**Review by jvigliotta — CHANGES_REQUESTED** [review]
This looks great! Can we change the double quotes to single quotes for the message (we're trying to stick to this moving forward)? Also, can we update the config to add the new option, but default it to false.

File: src/historical/HistoricalProvider.js [hunk 1]
```diff
@@ -522,6 +522,12 @@ define([
         if (session) {
             params.filter.session_id = '(' + session.numbers.join(',') + ')';
             params.filter.session_host = session.host;
+        } else if (window.openmctMCWSConfig?.sessions?.historicalSessionFilter?.denyUnfilteredQueries === true) {
```
Comment (davetsay) [inline src/historical/HistoricalProvider.js:525]:
```suggestion
        } else if (window.openmctMCWSConfig?.sessions?.historicalSessionFilter?.disable !== true && window.openmctMCWSConfig?.sessions?.historicalSessionFilter?.denyUnfilteredQueries === true) {
```

**Review by davetsay — CHANGES_REQUESTED** [review]
Hi @chesterhawkins . thanks so much for contributing. I have one more slight change. I added a code suggestion to also check for the `disable` prop not being true. If it is false we shouldn't care about the `denyUnfilteredQueries` prop.

**Review by davetsay — APPROVED** [review]
Awesome. Thanks, @chesterhawkins

**Review by jvigliotta — APPROVED** [review]
LGTM!

**Pull request merged**